client/boot/index.js

```diff
@@ -369,6 +370,30 @@ function reduxStoreReady( reduxStore ) {
 		require( 'lib/rubberband-scroll-disable' )( document.body );
 	}
 
+	// 404 handler.
+	page( '*', function( context, next ) {
+		let Page404 = require( 'layout/404.jsx' );
+
+		if ( ! config.isEnabled( 'code-splitting' ) ) {
+			// If code-splitting is not enabled and we got to this route handler, no section route was matched.
+			Page404.show();
```

Comment:
Yep, `client/layout/error.jsx` is the only way we provide static methods in all codebase. I wouldn't follow that route. You can alway create reusable functions that allow to achieve the same goal.